dash/test/AtomicTest.cc

```diff
@@ -223,3 +225,53 @@ TEST_F(AtomicTest, AtomicInterface){
   }
   array.barrier();
 }
+
+TEST_F(AtomicTest, MutexInterface){
+  dash::Mutex mx;
+  
+  dash::Shared<int> shared(dash::team_unit_t{0});
+  
+  if(dash::myid() == 0){
+    shared.set(0);
+  }
+  
+  mx.lock();
+  int tmp = shared.get();
+  shared.set(tmp + 1);
+  LOG_MESSAGE("Before %d, after %d", tmp, static_cast<int>(shared.get()));
+  // I guess here a flush is required, blocked by issue 322
+  mx.unlock();
+  
+  dash::barrier();
+  
+  while(!mx.try_lock()){
+    std::this_thread::sleep_for(std::chrono::microseconds(100));
```

Comment:
Since the critical region is quite short, I would avoid the sleep to increase pressure on the lock. 

A sleep may be interesting between the `get()` and the `set()` above to better detect if the lock actually works.